pkg/helper/rolepolicybindings/helpers.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rolepolicybindings
+
+import (
+	"fmt"
+	"strings"
+
+	"slices"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+
+	"github.com/openshift/rosa/pkg/aws"
+)
+
+const (
+	RolePolicyBindingFailedStatus = "failed"
+)
+
+func ValidateRolePolicyBindings(bindings *cmv1.RolePolicyBindingList) error {
+	for _, binding := range bindings.Slice() {
+		if binding.Status().Value() == RolePolicyBindingFailedStatus {
+			return fmt.Errorf("Failed to get attach policies of role %s: %s",
+				binding.Name(), binding.Status().Description())
+		}
+	}
+	return nil
+}
+
+func CheckMissingRolePolicyBindings(desired, actual *cmv1.RolePolicyBindingList) (string, bool) {
+	output := ""
+	actualBindings := map[string][]string{}
+	for _, binding := range actual.Slice() {
+		roleBindings := []string{}
+		if binding.Policies() != nil {
+			for _, policy := range binding.Policies() {
+				roleBindings = append(roleBindings, policy.Arn())
+			}
+		}
+		actualBindings[binding.Name()] = roleBindings
+	}
+	missingBindings := map[string][]string{}
+	for _, binding := range desired.Slice() {
+		for _, policy := range binding.Policies() {
```

Comment:
Can binding be nil?

Reply:
I believe as long as clusters have roles associated (which is required during cluster create), desired will never be nil.

Reply:
If they delete some bindings from AWS, but not the installer we can still sync right? It might be possible to be nil I think, it is quite cumbersome for a customer to reach such state :thinking:

Reply:
the "RolePolicyBindingList" used here is a list of roles which also has a list of policies (if retrievable via AWS).
Since all clusters have roles associated, the list of roles or bindings will always have a value.

To clarify, a binding here is an object that has an array of policies within it. Each object associates to each role of the cluster. A role always exists therefore, a binding object will always exist.

If the permissions of the installer role was inaccessible / not assumable for some reason, the role policy binding list being passed here would still be pointing to a list of bindings (not nil).

Although I personally think the impact is minimal, I'd say it is always safer to have a condition to always check for nils; if binding is nil.
if binding != nil proceed with the loop in 58.
Also in line 49.

Reply:
added the check to avoid possible edge case that binding is nil